src/GovUk.Education.ExploreEducationStatistics.Content.Model/User.cs

```diff
@@ -12,6 +12,8 @@ public class User
 
         public string Email { get; set; }
 
+        public DateTime? Deleted { get; set; }
```

Comment:
What's the relevance of this being a `DateTime` as opposed to a boolean? It looks like the conditions used with this property only check for nulls.

Reply:
It's for audit purposes - since it's cheap to make it a date instead of a boolean, we get the additional info of when the deletion happened should we ever need it.

To be honest, I wouldn't have any problem with this being a boolean, but generally in this kind of situation on EES we have used nullable datetimes. I think this might actually originated from a PR comment on one of my PRs, where I had used a boolean I got the request to turn it into a datetime.

Reply:
Agree with Mark, we've got another example of this in `ResourceRole`. https://github.com/dfe-analytical-services/explore-education-statistics/blob/c7410430c26672e8a97fc73688ed8d631a08366a/src/GovUk.Education.ExploreEducationStatistics.Content.Model/ResourceRole.cs#L35.

I'm going to leave another comment elsewhere about this though as it looks like the deletion request can be repeated and that in that case nothing is checking the user entity is not already flagged as deleted, in which case the date would just be updated, rather than an error status being returned.

By the way, we've also got this interface which if used and the entity is marked for deletion, the value will be set rather than the entity being deleted. Not sure if this approach was originally considered or not?

https://github.com/dfe-analytical-services/explore-education-statistics/blob/8f8f4d080b97bcc8334205bb971ee69b212c36b3/src/GovUk.Education.ExploreEducationStatistics.Common/Model/Timestamps.cs#L15-L18

Reply:
I wasn't aware that `ISoftDeletedTimestamp` existed. I'll switch to using it - seems preferable in the case that someone tries to remove a `User` entry somewhere else.

Reply:
Since I've added DeletedById, I've now backed out of this change.